internal/mailer/mailer.go

```diff
@@ -64,23 +64,18 @@ func withDefault(value, defaultValue string) string {
 	return value
 }
 
-func getSiteURL(referrerURL, siteURL, filepath, fragment string) (string, error) {
-	baseURL := siteURL
-	if filepath == "" && referrerURL != "" {
-		baseURL = referrerURL
-	}
-
-	site, err := url.Parse(baseURL)
-	if err != nil {
-		return "", err
-	}
+func getPath(filepath, params string) (*url.URL, error) {
```

Comment:
We just return the path + params here instead of the entire URL. Since the hostname in the URL is dependent on the custom domain, we need to pass the hostname down from the request and verify that it's a valid hostname in the allow list.